src/categorical_algebra/FinSets.jl

```diff
@@ -258,7 +258,11 @@ the form ``{1,...,n}``.
   func::V
   codom::Codom
 end
-
+function Base.convert(::Type{SetFunctionCallable},S::FinDomFunctionVector)
+  FinDomFunction((i->S.func[i]),
+                        FinSet(length(S.func)),
+                        S.codom)
+end
```

Comment:
Shouldn't have this conversion. Instead, construct vectors with explicit types to avoid overly strong typing.